src/core/evaluator.js

```diff
@@ -1372,9 +1372,13 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
         }
         cmap = cmap.getMap();
         // Convert UTF-16BE
-        // NOTE: cmap can be a sparse array, so use forEach instead of for(;;)
-        // to iterate over all keys.
-        cmap.forEach(function(token, i) {
+        // NOTE: cmap can be a sparse array, so use Object.keys to get all keys
+        // and iterate over it. Neither for(;;) nor forEach is used because they
+        // are slow when the cmap contains a large key.
+        var cmapKeys = Object.keys(cmap);
+        for (var i = 0, ii = cmapKeys.length; i < ii; ++i) {
+          var charCode = cmapKeys[i];
+          var token = cmapKeys[charCode];
```

Comment:
Shouldn't this be `cmap[charCode]`? Currently https://cdn.gonitro.com/img/primo/PrimoPDF_V5_User_Guide.pdf breaks with this patch and I think it is due to this line.